nomad/job_endpoint.go

```diff
@@ -62,26 +83,6 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 	}
 	defer metrics.MeasureSince([]string{"nomad", "job", "register"}, time.Now())
 
-	// Validate the arguments
-	if args.Job == nil {
-		return fmt.Errorf("missing job for registration")
-	}
-
-	// Initialize the job fields (sets defaults and any necessary init work).
-	canonicalizeWarnings := args.Job.Canonicalize()
-
-	// Add implicit constraints
-	setImplicitConstraints(args.Job)
-
-	// Validate the job and capture any warnings
-	err, warnings := validateJob(args.Job)
-	if err != nil {
-		return err
-	}
-
-	// Set the warning message
-	reply.Warnings = structs.MergeMultierrorWarnings(warnings, canonicalizeWarnings)
-
 	// Check job submission permissions
 	if aclObj, err := j.srv.ResolveToken(args.AuthToken); err != nil {
```

Comment:
We need to canonicalize the job before we can run it through permissions validation in some cases I think. (e.g resolving default NS,  and potentially some volumes stuff)